src/components/FollowButton/index.js

```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react';
+import './style.sass';
+import { isLoggedIn, getToken } from '../../utils/auth';
+import { api } from '../../utils/api';
+
+export class FollowButton extends Component {
+  static propTypes = {
+
+  }
+
+  state = {
+    following: false,
+    loading: false,
+  }
+
+  componentWillMount() {
+    // fetch logged in user following
+    if (isLoggedIn()) {
+      const user = getToken();
+      const { user: { username } } = user;
+
+      const { author } = this.props;
+      api.user.following(username)
+        .then((response) => {
+          const { data } = response;
+          const { following } = data;
+          // check if author is in following
+          const found = following.find(element => element.user__username === author.username);
+          if (found === undefined) {
+            this.setState({ following: true });
+          } else {
+            this.setState({ following: false });
+          }
+        })
+        .catch(err => console.warn(err));
+    }
+  }
+
+
+  followHandler = () => {
+    const { following } = this.state;
+    const { author } = this.props;
+    this.setState({ loading: true });
+
+    if (!following) {
+      // if am following the author (following == true)
```

Comment:
you do not need comments, the code should be self-explanatory

Reply:
okay